src/checkmatelib/client.py

```diff
@@ -68,10 +75,30 @@ def check_url(  # pylint: disable=inconsistent-return-statements
         response.raise_for_status()
 
         if response.status_code == 204:
+            # No news is good news
             return None
 
         try:
             return BlockResponse(response.json())
 
         except ValueError as err:
             raise CheckmateServiceError("Unprocessable JSON response") from err
+
+    @classmethod
+    def _clean_url(cls, url):
+        """Clean the URL before we send it.
+
+        Checkmate will do this for itself, and doesn't trust our input, but
+        this allows us to fail fast for bad URLs, before Checkmate has a chance
+        to check them. We can also apply extra constraints.
+        """
+
+        # Truncate extremely long URLs, so we don't get 400's and fail open
+        parts = CanonicalURL.canonical_split(url[: cls.MAX_URL_LENGTH])
+
+        # Enforce that domains are valid and public
+        domain = Domain(parts[1])
+        if domain not in cls.ALLOWED_PRIVATE_DOMAINS and not domain.is_public:
```

Comment:
To check my understanding of what `is_public` entails:

1. `domain` (which may in fact be an IP address) must contain a suffix that appears in https://publicsuffix.org/list/, and thus suffix must not be the whole value?
2. `domain` must not be an IPv4 address in the private range?

Are IPv6 addresses handled?

Reply:
Pretty much. The definition of what `is_public` means is actually mostly best explained by the code:

```python
    @property
    def is_public(self):
        """Get whether the domain is publicly available."""

        if self.is_ip_v4:
            return not self.is_private_ip_v4

        return self.is_fully_qualified
        
    @property
    @lru_cache(1)
    def is_fully_qualified(self):
        """Get whether the domain is a fully qualified domain name (public).

        IP addresses are not considered fully qualified, as they have no top
        level domain.
        """

        if not self.is_valid:
            return False

        # We have a proper suffix, and it's not the whole domain name
        return bool(self.icann_suffix) and (self != self.icann_suffix)
```

So we are public if we are:

 * A public IPV4 address
 * Or a valid domain name ...
 * ... which ends with an ICANN suffix
 * ... but is not entirely an ICANN suffix
 
I don't think IPV6 stuff is handled. I don't have any examples to check.